src/parquet/column/scanner-test.cc

```diff
@@ -0,0 +1,210 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include <cstdint>
+#include <cstdlib>
+#include <iostream>
+#include <sstream>
+#include <string>
+#include <vector>
+//#include <initializer_list>
+
+#include <gtest/gtest.h>
+
+#include "parquet/types.h"
+#include "parquet/column/page.h"
+#include "parquet/column/reader.h"
+#include "parquet/column/scanner.h"
+#include "parquet/column/test-util.h"
+
+#include "parquet/util/test-common.h"
+
+using parquet::FieldRepetitionType;
+using parquet::SchemaElement;
+
+namespace parquet_cpp {
+
+using schema::NodePtr;
+
+namespace test {
+
+struct PrimitiveColumn {
+  parquet_cpp::Type::type type;
+  std::vector<int32_t> values;
+  std::vector<int16_t> def_levels;
+  int width;
+  std::string printout;
+  PrimitiveColumn(parquet_cpp::Type::type type, const std::vector<int32_t>& values,
+      const std::vector<int16_t>& def_levels, int width, const std::string& printout) :
+    type(type), values(values), def_levels(def_levels),
+    width(width), printout(printout) {}
+};
+
+class TestPrimitiveScanner : public testing::TestWithParam<PrimitiveColumn> {
```

Comment:
I am not sure how `TestWithParam` helps here -- it makes the test fixture more rigid.